include/acquire.zarr.h

```diff
@@ -94,6 +94,32 @@ extern "C"
       const ZarrStreamSettings* settings,
       size_t* usage);
 
+    /**
+     * @brief Get the number of arrays configured in the Zarr stream settings,
+     * including both flat arrays and arrays in HCS plates.
+     * @param settings The Zarr stream settings struct.
+     * @return The number of array paths configured in the settings.
+     */
+    size_t ZarrStreamSettings_get_array_count(
+      const ZarrStreamSettings* settings);
+
+    /**
+     * @brief Get the array keys configured in the Zarr stream settings.
+     * The caller is responsible for freeing the memory allocated for each path
+     * in the array of paths.
+     * @param settings  The Zarr stream settings struct.
+     * @param[in, out] paths Pointer to an array of strings to be allocated and
+     * filled with the array paths. The caller is responsible for freeing the
+     * memory allocated for each path in the array.
+     * @param path_count Pointer to a size_t to be filled with the number of
+     * paths allocated.
+     * @return ZarrStatusCode_Success on success, or an error code on failure.
+     */
+    ZarrStatusCode ZarrStreamSettings_get_array_keys(
+      const ZarrStreamSettings* settings,
+      char*** paths,
```

Comment:
Triple pointers are a sign that the output data structure is too complex. Better might be something that just returns one key at a time

```
ZarrStatusCode ZarrStreamSettings_get_array_key(
      const ZarrStreamSettings* settings, size_t index,
      char** path);
```